### PR TITLE
Add Unicon Language Server Protocol Support

### PR DESCRIPTION
**Pull request by mstow597**
This feature is built in accordance to the Language Server Protocol (LSP).
Sources: 
- General overview of LSP: https://microsoft.github.io/language-server-protocol/
- LSP Specification: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/

Description:
Added new package, ulsp, to unicon/uni directory. Currently utilizing VS Code client for testing/debugging. The VS Code client implementation can be found at: https://github.com/HereBeCode/unicon-lsp

Current LSP Capabilities:
- textDocument/Synchronization
- textDocument/Completion

Future LSP Capabilities:
- Additional capabilities to be implemented can be found here: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#languageFeatures

### CODE REVIEW

File: uni/ulsp/Makefile [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# 	unicon -u -c server.icn
+
+# database.u:database.icn
+# 	unicon -u -c database.icn
+
+# completion.u: completion.icn
+# 	unicon -u -c completion.icn
+
+# clean:
+# 	rm -f *.u launch-lsp uniclass*
```
Comment (Jafaral) [inline uni/ulsp/Makefile:59]: Should we remove this?

File: uni/ulsp/database.icn [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    method build_package_db()
+
+	    paths := []
+	    put(paths, "/home/mark/Documents/unicon/uni/3d")
+		put(paths, "/home/mark/Documents/unicon/uni/gui")
+		#put(paths, "/home/mark/Documents/unicon/uni/parser")
+		put(paths, "/home/mark/Documents/unicon/uni/xml")
+		put(paths, "/home/mark/Documents/unicon/uni/lib")
+		put(paths, "/home/mark/Documents/unicon/ipl/procs")
+		put(paths, "/home/mark/Documents/unicon/uni/unidoc")
```
Comment (Jafaral) [inline uni/ulsp/database.icn:47]: path is still pointing at your local environment

**Review by Jafaral — CHANGES_REQUESTED** [review]
A quick couple of review points, more to come later

File: uni/ulsp/Makefile (hunk 1 above)
Comment (mstow597) [reply uni/ulsp/Makefile]: Removed comments
Related commit: https://github.com/uniconproject/unicon/pull/305/commits/3354f020e827e3eee006f7ef02998c24b8b67048

File: uni/ulsp/database.icn (hunk 2 above)
Comment (mstow597) [reply uni/ulsp/database.icn]: Paths now relative to unicon directory
Related commit: https://github.com/uniconproject/unicon/pull/305/commits/979d763d00190a6d75441114c78588f5faffa372

File: uni/ulsp/completion.icn [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+		#-----------------------------------------------#
+
+		if desired_line = &null return &null
+
+		#-----------------------------------------------#
+		#----------- Case "inside comment" -------------#
+		#-----------------------------------------------#
+
+		desired_line ? {
+			while (&pos < character) do {
```
Comment (Jafaral) [inline uni/ulsp/completion.icn:98]: Why loop? can't we just use `upto()`?

desired_line ? if upto('#') then return "inside comment"

**Review by Jafaral — CHANGES_REQUESTED** [review]
Thanks Mark! Another round of comments.

**Review by Jafaral — APPROVED** [review]

**Pull request merged**